src/transformers/models/falcon/modeling_falcon.py

```diff
@@ -519,6 +543,253 @@ def forward(
                 return output_tensor, present
 
 
+class FalconFlashAttention2(nn.Module):
```

Comment:
(likely controversial) should this just inherit from `FalconAttention` and redefine the `forward`?

Reply:
Yes agreed!